public/Invoke-DbaQuery.ps1

```diff
@@ -394,15 +411,74 @@ function Invoke-DbaQuery {
+                    # to Microsoft.Data.SqlClient.SqlConnection it enables us to leverage [Microsoft.Data.SqlClient.SqlParameter]s !
+                    #
+                    # Again, here we are, but we cannot use the connection when an information is lost, which is that we are:
+                    #   - Integrated Security=True
+                    #   - using ConnectAsUser, ConnectAsUserName, ConnectAsUserPassword to "log on as a different user"
+
+                    if ($instance.InputObject.ConnectionContext.ConnectAsUserName -ne '') {
+                        Write-Message -Level Debug -Message "Current connection cannot be reused because logging in as a different user"
+                        # We rebuild correct credentials from SMO informations
+                        $secStringPassword = ConvertTo-SecureString $instance.InputObject.ConnectionContext.ConnectAsUserPassword -AsPlainText -Force
+                        [PSCredential]$serverCredentialFromSMO = New-Object System.Management.Automation.PSCredential($instance.InputObject.ConnectionContext.ConnectAsUserName, $secStringPassword)
+                        $connDbaInstanceParams = @{
+                            SqlInstance            = $instance
+                            SqlCredential          = $serverCredentialFromSMO
+                            Database               = $Database
+                            NonPooledConnection    = $true           # see #8491 for details, also #7725 is still relevant
+                            Verbose                = $false
+                            AppendConnectionString = $AppendConnectionString
+                        }
+                        if ($ReadOnly) {
+                            $connDbaInstanceParams.ApplicationIntent = "ReadOnly"
+                        }
+
+                        $server = Connect-DbaInstance @connDbaInstanceParams
+
+                    } else {
+                        Write-Message -Level Debug -Message "Current connection will be reused"
+                        $server = $instance.InputObject
+                    }
+
                 } else {
                     $connDbaInstanceParams = @{
-                        SqlInstance         = $instance
-                        SqlCredential       = $SqlCredential
-                        Database            = $Database
-                        NonPooledConnection = $true           # see #8491 for details, also #7725 is still relevant
-                        Verbose             = $false
+                        SqlInstance            = $instance
+                        SqlCredential          = $SqlCredential
+                        Database               = $Database
+                        NonPooledConnection    = $true           # see #8491 for details, also #7725 is still relevant
+                        Verbose                = $false
+                        AppendConnectionString = $AppendConnectionString
```

Comment:
Sorry, @niphlod - this line triggers the following warning:
```
WARNING: [21:47:06][Connect-DbaInstance] Additional parameters are passed in, but they will be ignored
```

If $instance is already a Server SMO, then Connect-DbaInstance will reuse it and ignore $AppendConnectionString. We would have to test if $AppendConnectionString is used and in this case open a new connection.

To test:
```
$server = Connect-DbaInstance -SqlInstance SQL2022
Invoke-DbaQuery -SqlInstance $server -Database tempdb -Query "select 1"
```